File: go/09_logging_failures.go
// A09 - Logging Failures. VULN: no structured security audit log.
package main

import "log"

// VULN: A09 - failed admin actions not logged
func SetRole(userID, role string) {
	log.Printf("role change %s -> %s", userID, role) // no severity, no alert
}

File: python/09_logging_failures.py
"""A09 - Logging Failures. VULN: logs secrets + no security events."""
import logging

log = logging.getLogger("app")

# VULN: A09 - password logged; no failed-login audit
def authenticate(username, password):
    log.info("login attempt user=%s password=%s", username, password)
    return username == "admin" and password == "admin"
